dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxTakeProfitStopLossInputAreaViewModel.swift

```diff
@@ -0,0 +1,208 @@
+    )
+
+    public static var previewValue: dydxTakeProfitStopLossInputAreaModel = {
+        let vm = dydxTakeProfitStopLossInputAreaModel()
+        vm.gainInputViewModel = dydxGainLossInputViewModel(triggerType: .takeProfit)
+        vm.lossInputViewModel = dydxGainLossInputViewModel(triggerType: .stopLoss)
+        vm.takeProfitPriceInputViewModel = dydxTriggerPriceInputViewModel(triggerType: .takeProfit)
+        vm.stopLossPriceInputViewModel = dydxTriggerPriceInputViewModel(triggerType: .stopLoss)
+        return vm
+    }()
+
+    private func createTooltip(triggerType: dydxTakeProfitStopLossInputAreaModel.TriggerType) -> some View {
+        guard let localizedString = DataLocalizer.shared?.localize(path: triggerType.sectionTitleLocalizerPath, params: nil) else { return EmptyView().wrappedInAnyView() }
+        let attributedTitle = AttributedString(localizedString)
+            .themeFont(fontSize: .medium)
+            .themeColor(foreground: .textSecondary)
+
+        let binding: Binding<Bool>
+        switch triggerType {
+        case .takeProfit:
+            binding = self.isTakeProfitTooltipPresentedBinding
+        case .stopLoss:
+            binding = self.isStopLossTooltipPresentedBinding
+        }
+
+        return Text(attributedTitle.dottedUnderline(foreground: .textSecondary))
+            .onTapGesture { [weak self] in
+                switch triggerType {
+                case .takeProfit:
+                    self?.isTakeProfitTooltipPresented.toggle()
+                case .stopLoss:
+                    self?.isStopLossTooltipPresented.toggle()
+                }
+            }
+            .popover(present: binding, attributes: {
+                $0.position = .absolute(
+                      originAnchor: .top,
+                      popoverAnchor: .bottom
+                  )
+                $0.sourceFrameInset = .init(top: 0, left: 0, bottom: -16, right: 0)
+                $0.presentation.animation = .none
+                $0.blocksBackgroundTouches = true
+                $0.onTapOutside = {
+                    switch triggerType {
+                    case .takeProfit:
+                        self.isTakeProfitTooltipPresented = false
+                    case .stopLoss:
+                        self.isStopLossTooltipPresented = false
+                    }
+                }
+            }, view: {
+                Text(localizerPathKey: triggerType.sectionTitleTooltipLocalizerPath)
+                    .padding(.vertical, 10)
+                    .padding(.horizontal, 12)
+                    .themeFont(fontType: .base, fontSize: .small)
+                    .themeColor(foreground: .textSecondary)
+                    .themeColor(background: .layer5)
+                    .borderAndClip(style: .cornerRadius(8), borderColor: .layer6, lineWidth: 1)
+                    .environmentObject(ThemeSettings.shared)
+            })
+            .wrappedInAnyView()
```

Comment:
i created a ticket to make this a viewmodifier

[MOB-438 : refactor tool tip to view modifier](https://linear.app/dydx/issue/MOB-438/refactor-tool-tip-to-view-modifier)